src/main.rs

```diff
@@ -16,142 +18,149 @@ fn create_static_sampler(
-                u: IRRootParameter1_u {
-                    constants: IRRootConstants {
-                        register_space: 1 as u32,
-                        shader_register: 0,
-                        num32_bit_values: 1,
-                    },
-                },
-            };
-
-            vec![push_constants, indirect_identifier]
-        };
+        // Create an instance of IRCompiler
+        let compiler_factory = IRCompilerFactory::new("libmetalirconverter.dylib").unwrap();
+        let mut compiler = compiler_factory.create_compiler();
 
-        let static_samplers = [
-            create_static_sampler(
-                IRFilter::IRFilterMinMagMipPoint,
-                IRTextureAddressMode::IRTextureAddressModeWrap,
-                0,
-                None,
-            ),
-            create_static_sampler(
-                IRFilter::IRFilterMinMagMipPoint,
-                IRTextureAddressMode::IRTextureAddressModeClamp,
-                1,
-                None,
-            ),
-            create_static_sampler(
-                IRFilter::IRFilterMinMagMipLinear,
-                IRTextureAddressMode::IRTextureAddressModeWrap,
-                2,
-                None,
-            ),
-            create_static_sampler(
-                IRFilter::IRFilterMinMagMipLinear,
-                IRTextureAddressMode::IRTextureAddressModeClamp,
-                3,
-                None,
-            ),
-            create_static_sampler(
-                IRFilter::IRFilterMinMagMipLinear,
-                IRTextureAddressMode::IRTextureAddressModeBorder,
-                4,
-                None,
-            ),
-            create_static_sampler(
-                IRFilter::IRFilterAnisotropic,
-                IRTextureAddressMode::IRTextureAddressModeWrap,
-                5,
-                Some(2),
-            ),
-            create_static_sampler(
-                IRFilter::IRFilterAnisotropic,
-                IRTextureAddressMode::IRTextureAddressModeWrap,
-                6,
-                Some(4),
-            ),
-        ];
+        // Create an explicit root signature layout
+        let mut parameters = create_root_parameters();
+        let mut static_samplers = create_static_samplers();
 
         let desc_1_1 = IRRootSignatureDescriptor1 {
-            flags: IRRootSignatureFlags::IRRootSignatureFlagCBVSRVUAVHeapDirectlyIndexed,
-            num_parameters: parameters.len() as u32,
-            p_parameters: parameters.as_ptr(),
-            num_static_samplers: static_samplers.len() as u32,
-            p_static_samplers: static_samplers.as_ptr(),
+            Flags: IRRootSignatureFlags::IRRootSignatureFlagCBVSRVUAVHeapDirectlyIndexed,
+            NumParameters: parameters.len() as u32,
+            pParameters: parameters.as_mut_ptr(),
+            NumStaticSamplers: static_samplers.len() as u32,
+            pStaticSamplers: static_samplers.as_mut_ptr(),
```

Comment:
This is for example why I think we should export `ffi` types behind a module, so that we can provide more high-level rusty wrappers to callers in the future.